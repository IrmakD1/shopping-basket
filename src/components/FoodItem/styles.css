.item-box {
    padding: 0.5rem;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    border: 2px solid #5171b6;
    border-radius: 3px;
    max-width: 17rem;
}

img {
    height: 30px;
    margin-right: 1rem;
    margin: 0.5rem
}

.plus-button {
    height: 25px;
    margin-top: 0.7rem;
    margin-left: 0.5rem;
    border: 0.1em solid #5171b6;
    border-radius: 0.12em;
    font-weight: bolder;
}

.minus-button-display-false {
    display: none;
}

.minus-button-display-true {
    height: 25px;
    margin-top: 0.7rem;
    border: 0.1em solid #5171b6;
    border-radius: 0.12em;
    font-weight: bolder;
}