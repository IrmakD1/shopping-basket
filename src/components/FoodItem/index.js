import React from 'react'
import './styles.css'


const FoodItem = ({ item, handleClick, handleDelete, total, totalCost, display }) => {
    return (
        <div className='item-box'>
            <div className='food-icon'>
                <img src={item.img} alt={`${item.item} icon`}/>
            </div>
            <p>{`${total} ${item.unit} of ${item.item}: ${item.currencyUnit}${totalCost}`}</p>
            <button className='plus-button' onClick={() => handleClick(item)}> + </button>
            <button className={`minus-button-display-${display}`} onClick={() => handleDelete(item)}> - </button>
        </div>
    )
}

FoodItem.defaultProps = {
    total: '',
    display: false
}

export default FoodItem