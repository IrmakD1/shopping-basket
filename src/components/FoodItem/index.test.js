import React from 'react';
import renderer from 'react-test-renderer';
import FoodItem from '.';

const item = {
    unit: 'slice',
    img: 'test',
    item: 'Kevin Bacon',
    currencyUnit: '$',
}

const handleClick = jest.fn()

const handleDelete = jest.fn()


describe('FoodItem Component on Food page', () => {
    let component;
    let tree;
  
    beforeEach(() => {
      component = renderer.create(
        <div>
            <FoodItem 
                item={item}
                handleClick={handleClick}
                totalCost={1}
            />
        </div>
      );
      tree = component.toJSON();
    });
  
    test('it should render', () => {
      expect(tree).toMatchSnapshot();
    });
});

describe('FoodItem Component on basket page', () => {
    let component;
    let tree;
  
    beforeEach(() => {
      component = renderer.create(
        <div>
            <FoodItem 
                item={item}
                handleClick={handleClick}
                totalCost={1}
                total={2}
                display={true}
            />
        </div>
      );
      tree = component.toJSON();
    });
  
    test('it should render', () => {
      expect(tree).toMatchSnapshot();
    });
});
  