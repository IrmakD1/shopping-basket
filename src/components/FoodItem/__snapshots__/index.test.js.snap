// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodItem Component on Food page it should render 1`] = `
<div>
  <div
    className="item-box"
  >
    <div
      className="food-icon"
    >
      <img
        alt="Kevin Bacon icon"
        src="test"
      />
    </div>
    <p>
       slice of Kevin Bacon: $1
    </p>
    <button
      className="plus-button"
      onClick={[Function]}
    >
       + 
    </button>
    <button
      className="minus-button-display-false"
      onClick={[Function]}
    >
       - 
    </button>
  </div>
</div>
`;

exports[`FoodItem Component on basket page it should render 1`] = `
<div>
  <div
    className="item-box"
  >
    <div
      className="food-icon"
    >
      <img
        alt="Kevin Bacon icon"
        src="test"
      />
    </div>
    <p>
      2 slice of Kevin Bacon: $1
    </p>
    <button
      className="plus-button"
      onClick={[Function]}
    >
       + 
    </button>
    <button
      className="minus-button-display-true"
      onClick={[Function]}
    >
       - 
    </button>
  </div>
</div>
`;
